go.work

```diff
@@ -0,0 +1,7 @@
+go 1.19
```

Comment:
Supposedly you don't want to check this in.  I'd suggest avoiding multiple modules if you can, but /shrug

Reply:
isolating multiple modules used to be more important for keeping dependency trees manageable (see also tools module pattern). I think that's no longer necessary, tidy seems to be getting smarter.

Reply:
I started working on this a while back, this is one of the errant bits I should clean up before merge.

The PR _is_ titled WIP and the description mentions it's just for visibility, and not ready yet :-)

Reply:
Seems go1.17 is when this got fixed, and only for modules that declare themselves as 1.17 or higher https://go.dev/ref/mod#graph-pruning

Reply:
keeping the tools module for now because the dep graph is messier and _only_ used at test time (and we still have to have no conflicts between those version and the libraries we _actually_ import if we drop that module), but dropped the geranos module and go.work